### PR TITLE
ref(eventstream): Use confluent-kafka

### PR DESCRIPTION
**Pull request by tkaemming**
- Uses the librdkafka-based Confluent Kafka producer for consistency moving forward
- Removes async/sync option

This will need to be synchronized with a dependency change and pin in getsentry/getsentry as well as a configuration update to use the librdkafka configuration mapping.

### CODE REVIEW

File: src/sentry/eventstream/kafka.py
Comment (tkaemming) [reply src/sentry/eventstream/kafka.py]: (Calling `flush` would also obviate my long-winded comment about `poll` above as well.)

**Comment by tkaemming** [comment]
Also, it'd be really nice here to have integration tests as this moves onto the critical path

File: src/sentry/eventstream/kafka.py
Comment (bretthoerner) [inline src/sentry/eventstream/kafka.py]: The process running this code is doing a tight-loop that's processing events, right? If so, I think the `poll` is probably fine. But if this ever moved to happen in a web process that could `publish` and then do 30 minutes of random API requests it would get pretty weird and I'd agree with the extra thread.

**Review by bretthoerner — APPROVED** [review]
Looks right to me but I'd say we definitely want another pair of eyes on this bad boyyy.

**Comment by bretthoerner** [comment]
Oh, and maybe you noted it somewhere but this is blocked on `librdkafka` being available in production ya?

File: src/sentry/eventstream/kafka.py
Comment (tkaemming) [reply src/sentry/eventstream/kafka.py]: I think the webs are handling both ingest as well as API, unless anything has changed recently?

**Comment by tkaemming** [comment]
> Oh, and maybe you noted it somewhere but this is blocked on `librdkafka` being available in production ya?

I noted the Python dependency in the description but hadn't considered the `librdkafka` library specifically, good point

File: src/sentry/eventstream/kafka.py
Comment (bretthoerner) [reply src/sentry/eventstream/kafka.py]: I thought webs did that first-pass method, then we did the whole Rabbit/Redis dance, and actual event processing (so including `eventstream` publish) happened on Celery workers?

Comment (tkaemming) [reply src/sentry/eventstream/kafka.py]: Oops, yeah — you're right, ignore me

**Pull request merged**
